File: MPI/mpicheck/tests/functions/4.0/MPI_Iallgatherv_c.f90

        
        program main
        use mpi
        
        TYPE(INTEGER) var_0(10)
       INTEGER(KIND=MPI_COUNT_KIND) var_1
       INTEGER var_2
       TYPE(INTEGER) var_3(10)
       INTEGER(KIND=MPI_COUNT_KIND) var_4(10)
       INTEGER(KIND=MPI_ADDRESS_KIND) var_5(10)
       INTEGER var_6
       INTEGER var_7
       INTEGER var_8
       INTEGER var_9
        call mpi_iallgatherv_c(var_0, var_1, var_2, var_3, var_4, var_5, var_6, var_7, var_8, &
       var_9)
       call pmpi_iallgatherv_c(var_0, var_1, var_2, var_3, var_4, var_5, var_6, var_7, var_8, &
       var_9)
        end program main
    

File: MPI/mpicheck/tests/functions/4.0/MPI_Iallgatherv_c.c
#include <mpi.h>
int main(char argc, char**argv)
{
    /* vars */
    const void *var_0;
    MPI_Count var_1;
    MPI_Datatype var_2;
    void *var_3;
    const MPI_Count var_4[2];
    const MPI_Aint var_5[2];
    MPI_Datatype var_6;
    MPI_Comm var_7;
    MPI_Request *var_8;
    int ret;
    /* calls */
    ret = MPI_Iallgatherv_c(var_0, var_1, var_2, var_3, var_4, var_5, var_6, var_7, var_8);
    ret = PMPI_Iallgatherv_c(var_0, var_1, var_2, var_3, var_4, var_5, var_6, var_7, var_8);
    return 0;
}
